src/Microsoft.AspNet.Mvc.Razor/TagHelpers/UrlResolutionTagHelper.cs

```diff
@@ -0,0 +1,206 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Reflection;
+using Microsoft.AspNet.Mvc.Rendering;
+using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+using Microsoft.Framework.WebEncoders;
+
+namespace Microsoft.AspNet.Mvc.Razor.TagHelpers
+{
+    /// <summary>
+    /// <see cref="ITagHelper"/> implementation targeting elements containing attributes with url expected values.
+    /// </summary>
+    /// <remarks>Resolves application relative URLs.</remarks>
```

Comment:
Also mention that this runs prior to other tag helpers and thus targets elements and attributes based on the Razor file content. Plus that this tag helper only targets attributes no other tag helper binds.